src/rp2_common/pico_stdio_usb/reset_interface.c

```diff
@@ -38,7 +38,10 @@ static uint16_t resetd_open(uint8_t __unused rhport, tusb_desc_interface_t const
 }
 
 // Support for parameterized reset via vendor interface control request
-static bool resetd_control_request_cb(uint8_t __unused rhport, tusb_control_request_t const *request) {
+static bool resetd_control_xfer_cb(uint8_t rhport, uint8_t stage, tusb_control_request_t const * request) {
+    // nothing to do with DATA & ACK stage
+    if (stage != CONTROL_STAGE_SETUP) return true;
```

Comment:
Since you are reset mcu, it should be better to reset when `stage == ACK`  (ACK is complete), this will prevent host OS to complain about the failure of control transfer. For example,  touch1200 is common way for Arduino to do reset to DFU. 

https://github.com/hathach/tinyusb/blob/master/src/class/cdc/cdc_device.c#L354
```C
    case CDC_REQUEST_SET_LINE_CODING:
      if (stage == CONTROL_STAGE_SETUP)
      {
        TU_LOG2("  Set Line Coding\r\n");
        tud_control_xfer(rhport, request, &p_cdc->line_coding, sizeof(cdc_line_coding_t));
      }
      else if ( stage == CONTROL_STAGE_ACK)
      {
        if ( tud_cdc_line_coding_cb ) tud_cdc_line_coding_cb(itf, &p_cdc->line_coding);
      }
    break;
```

Reply:
ah; good point thanks